pkg/services/ngalert/eval/eval.go

```diff
@@ -379,13 +380,28 @@ func queryDataResponseToExecutionResults(c models.Condition, execResp *backend.Q
 
 		if len(frame.Fields) == 1 {
 			theseLabels := frame.Fields[0].Labels
-			for _, cap := range captures {
-				// matching labels are equal labels, or when one set of labels includes the labels of the other.
-				if theseLabels.Equals(cap.Labels) || theseLabels.Contains(cap.Labels) || cap.Labels.Contains(theseLabels) {
-					if frame.Meta.Custom == nil {
-						frame.Meta.Custom = []NumberValueCapture{}
+			fp := theseLabels.Fingerprint()
+
+			// First look for a capture whose labels are an exact match
+			if groupedCaps, ok := captures[fp]; ok {
```

Comment:
As this branch didn't exist before, I would recommend double checking that both of these cases are exercised by the unit test (exact matches and subset matches).